### DISCUSSION TITLE
Why tslib as dependency?

### DISCUSSION THREAD

**Discussion by atassis** [ReactiveX/rxjs#5969] [Q&A]
You shouldn't rely on tslib at all, it is a typescript compilation level helper, as documented on its side. I saw that by checking dependencies on my server code. We don't use `importHelpers` flag, it is totally unnecessary in any kind of environment, but still its the single dependency you have
 
**RxJS version:**
6.x and higher

**Expected behavior:**
tslib not being listed as dependency at all

**Actual behavior:**
tslib as dependency

**Additional information:**
https://github.com/Microsoft/tslib

**Comment by kwonoj** [comment]
rxjs itself currently uses importhelpers https://github.com/ReactiveX/rxjs/blob/9e9064b7513aec4c365e792beac211f9d12bd7b9/src/tsconfig.esm.json#L5 and try to dedupe runtime code generated by tsc per each files. There are some internal discussion around if we should get away from it or not after re-do size benchmark. 

It was introduced due to benefit of tslib

> this can avoid duplicate declarations of things like __extends, __assign, etc., this means delivering users smaller files on average, as well as less runtime overhead. For optimized bundles with TypeScript, you should absolutely consider using tslib and --importHelpers.

in lib itself. 

`it is a typescript compilation level helper`

I'm curios to hear more about this - tslib pkg itself says it's `This is a runtime library` for tsc builds and we use it for those purpose.

**Comment by kwonoj** [comment]
And I'm moving this into discussions as issues are for core team's backlog for bug tracking.

**Comment by benlesh** [comment]
I think by version 8 we might be able to remove this. Really we have this in place to support IE and older versions of Node.

**Comment by kwonoj** [comment]
One another reason at the moment was it reduced total bytesize by consolidating boilerplate. I do not know if that's still the case, and would like to remove tslib as well if possible.
